tests/5.0/tasks/test_task_depend_iterator.cpp

```diff
@@ -0,0 +1,56 @@
+//===-- test_task_depend_iterator.cpp - test of the iterator clause inside the task depend clause ----===//
+// 
+// OpenMP API Version 5.0 Nov 2020
+// 
+// This file is a test for the iterator modifier when used with the task depend
+// clause. This modifier should create an iterator that expand into multiple values
+// inside the clause they appear. In this particular test case the iterator expands into
+// several values creating several dependencies at the same time.
+//
+//===-------------------------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+#include <vector>
+#include "ompvv.h"
+
+int test_task_depend_iterator() {
+  int ptr[] = {0, 4, 5, 6, 7, 8, 9, 10, 11}, cols[] = {1, 2, 3, 4, 5, 5, 6, 6, 7, 7, 8};
+  std::vector<int> threadOrder;
+  bool threadOrderError = false;
+#pragma omp parallel num_threads(4)
+  {
+#pragma omp single
+    {
+      for (int i = 0; i < 8; ++i) {
+        int pos = ptr[i], size = ptr[i + 1] - ptr[i];
+#pragma omp task depend(iterator(it = 0 : size), in : ptr[cols[pos + it]]) depend(out : ptr[i])
+        {
+#pragma omp critical
+          {
+            threadOrder.push_back(i);
+          } // end critical section
+          std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
we don't need this sleep here.

Reply:
That was to try to ensure the task execution on different threads. However since I removed the thread assignment check it can be removed.

Reply:
Removed